### PR TITLE
Implemented C# version of mono_class_is_enum

### PR DESCRIPTION
**Pull request by jeanclaudegrenier**

### CODE REVIEW

**Review by UnityAlex — APPROVED** [review]

File: unity/UnityEmbedHost.Tests/BaseEmbeddingApiTests.cs [hunk 1]
```diff
@@ -616,6 +616,18 @@ static void AssertMultiDimensionalArraysAreEquivalent(Array expected, Array actu
         Assert.That(actualFlat, Is.EquivalentTo(expectedFlat));
     }
 
+    [TestCase(typeof(Classification),              true)]
+    [TestCase(typeof(Classification*),            false)]
+    [TestCase(typeof(Classification[]),           false)]
+    [TestCase(typeof(Animal),                     false)]
+    [TestCase(typeof(IAnimal),                    false)]
+    [TestCase(typeof(GenericAnimal<int, string>), false)]
```
Comment (joncham) [inline unity/UnityEmbedHost.Tests/BaseEmbeddingApiTests.cs]: Nit, one more test should be an enum inside as generic as this is a common corner case causing errors.

struct Foo<T>
{
    public enum Bar { One, Two };
}

**Review by joncham — APPROVED** [review]
